google/resource_compute_shared_vpc.go

```diff
@@ -150,7 +147,16 @@ func resourceComputeSharedVpcDelete(d *schema.ResourceData, meta interface{}) er
 	return nil
 }
 
-func enableResource(config *Config, hostProject, project string) error {
+func isDisabledXpnResourceError(err error) bool {
```

Comment:
Added `isDisabledXpnResourceError()` otherwise resource deletion (esp. for the Shared VPC resource) fails when attempting to disable an already disabled service project:
```
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: HTTP/2.0 400 Bad Request
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Alt-Svc: quic=":443"; ma=2592000; v="39,38,37,35"
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Cache-Control: private, max-age=0
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Content-Type: application/json; charset=UTF-8
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Date: Sat, 23 Sep 2017 21:37:26 GMT
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Expires: Sat, 23 Sep 2017 21:37:26 GMT
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Server: GSE
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Vary: Origin
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: Vary: X-Origin
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: X-Content-Type-Options: nosniff
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: X-Frame-Options: SAMEORIGIN
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: X-Xss-Protection: 1; mode=block
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: 
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: {
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:  "error": {
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:   "errors": [
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:    {
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:     "domain": "global",
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:     "reason": "invalidResourceUsage",
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:     "message": "Invalid resource usage: 'The resource 'projects/test-service4-id' is not linked to shared VPC host 'projects/test-host4-id'.'."
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:    }
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:   ],
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:   "code": 400,
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:   "message": "Invalid resource usage: 'The resource 'projects/test-service4-id' is not linked to shared VPC host 'projects/test-host4-id'.'."
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google:  }
2017-09-23T17:37:26.228-0400 [DEBUG] plugin.terraform-provider-google: }
```